### ISSUE TITLE
Change Koa variant from generator function to async one

### ISSUE THREAD

**Issue by garcia556** [wilmoore/request-id.js#9]
Here's what Koa says:

koa deprecated Support for generators will be removed in v3. See the documentation for examples of how to convert old middleware https://github.com/koajs/koa/blob/master/docs/migration.md

Would be nice to migrate it here in request-id too.

**Comment by wilmoore** [comment]
Good call. I'll handle it when I've got some free time; otherwise, I'll take a PR from those looking to contribute.
